mllib/src/main/scala/org/apache/spark/ml/r/IsotonicRegressionWrapper.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.r
+
+import org.apache.hadoop.fs.Path
+import org.json4s._
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods._
+
+import org.apache.spark.ml.{Pipeline, PipelineModel}
+import org.apache.spark.ml.attribute.{Attribute, AttributeGroup, NominalAttribute}
+import org.apache.spark.ml.feature.RFormula
+import org.apache.spark.ml.regression.{IsotonicRegression, IsotonicRegressionModel}
+import org.apache.spark.ml.util._
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+private [r] class IsotonicRegressionWrapper private (
+    val pipeline: PipelineModel,
+    val labels: Array[String],
+    val features: Array[String]) extends MLWritable {
+
+  private val isotonicRegressionModel: IsotonicRegressionModel =
+    pipeline.stages(1).asInstanceOf[IsotonicRegressionModel]
+
+  lazy val boundaries: Array[Double] = isotonicRegressionModel.boundaries.toArray
+
+  lazy val predictions: Array[Double] = isotonicRegressionModel.predictions.toArray
+
+  def fitted(method: String): Array[Double] = {
+    if (method == "boundaries") {
+      boundaries
+    } else if (method == "predictions") {
+      predictions
+    } else {
+      throw new UnsupportedOperationException(
+        s"Method (boundaries or predictions) required but $method found.")
+    }
+  }
+
+  def transform(dataset: Dataset[_]): DataFrame = {
+    pipeline.transform(dataset).drop(isotonicRegressionModel.getFeaturesCol)
+  }
+
+  override def write: MLWriter = new IsotonicRegressionWrapper.IsotonicRegressionWrapperWriter(this)
+}
+
+private[r] object IsotonicRegressionWrapper
+    extends MLReadable[IsotonicRegressionWrapper] {
+
+  def fit(
+      data: DataFrame,
+      formula: String,
+      isotonic: Boolean,
+      featureIndex: Int): IsotonicRegressionWrapper = {
+
+    val rFormulaModel = new RFormula()
+      .setFormula(formula)
+      .fit(data)
+
+    // get feature names from output schema
+    val schema = rFormulaModel.transform(data).schema
+    val labelAttr = Attribute.fromStructField(schema(rFormulaModel.getLabelCol))
+      .asInstanceOf[NominalAttribute]
+    val labels = labelAttr.values.get
```

Comment:
Since ```IsotonicRegression``` is a regression model, so it's unnecessary to extract labels from column metadata (actually we did not save ```NominalAtrribute``` for regression model). Thanks!